test_lib/sla.py

```diff
@@ -125,119 +213,89 @@ def name(self, name):
         self._name = name
 
     @property
-    def attached_service_level_name(self):
-        return self._attached_service_level_name
-
-    @attached_service_level_name.setter
-    def attached_service_level_name(self, service_level_name):
-        self._attached_service_level_name = service_level_name
+    def attached_service_level(self):
+        return self._attached_service_level
 
     @property
-    def attached_service_level_shares(self):
-        return self._attached_service_level_shares
-
-    @attached_service_level_shares.setter
-    def attached_service_level_shares(self, service_level_shares):
-        self._attached_service_level_shares = service_level_shares
+    def attached_service_level_name(self):
+        return self._attached_service_level.name
 
     def attach_service_level(self, service_level):
-        query = 'ATTACH SERVICE_LEVEL {service_level_name} TO {role_name}' \
-            .format(service_level_name=service_level.name,
-                    role_name=self.name)
+        """
+        :param auth_name: it may be role name or user name
+        """
+        query = f'ATTACH SERVICE_LEVEL {service_level.name} TO {self.name}'
         if self.verbose:
-            self.log.debug('Attach service level query: {}'.format(query))
+            LOGGER.debug('Attach service level query: {}'.format(query))
```

Comment:
Since you touching all these logging stuff, can you convert them to lazy-logging too, please?

Here and below.